lib/rspec-gc-control/base_text_formatter.rb

```diff
@@ -1,45 +1,50 @@
+require 'rspec/core/version'
 require 'rspec/core/formatters/base_formatter'
 
-module RSpec
-  module Core
-    module Formatters
+if defined?(::RSpec::Core) && ::RSpec::Core::Version::STRING < '3.0.0'
 
-      class BaseTextFormatter < BaseFormatter
+  module RSpec
+    module Core
+      module Formatters
 
-        # Overridden version of this method, to include output about forced GC
-        # count and duration, if explicit GC control is enabled.
-        def dump_summary(duration, example_count, failure_count, pending_count)
-          super(duration, example_count, failure_count, pending_count)
+        class BaseTextFormatter < BaseFormatter
 
-          gc_times  = examples.
-                        map { |ex| ex.execution_result[:gc_time] }.
-                        select { |t| t && (t > 0) }
-          gc_count  = gc_times.count
-          gc_time   = gc_times.reduce(:+)
+          # Overridden version of this method, to include output about forced GC
+          # count and duration, if explicit GC control is enabled.
+          def dump_summary(duration, example_count, failure_count, pending_count)
+            super(duration, example_count, failure_count, pending_count)
 
-          dump_summary_to_output(duration, gc_time, gc_count, example_count,
-                                 failure_count, pending_count)
-        end
+            gc_times  = examples.
+                          map { |ex| ex.execution_result[:gc_time] }.
+                          select { |t| t && (t > 0) }
+            gc_count  = gc_times.count
+            gc_time   = gc_times.reduce(:+)
+
+            dump_summary_to_output(duration, gc_time, gc_count, example_count,
+                                   failure_count, pending_count)
+          end
 
-        protected
+          protected
 
-        def dump_summary_to_output(duration, gc_time, gc_count, example_count,
-                           failure_count, pending_count)
-          # Don't print out profiled info if there are failures, it just
-          # clutters the output.
-          dump_profile if profile_examples? && failure_count == 0
-          output.print "\nFinished in #{format_duration(duration)}"
+          def dump_summary_to_output(duration, gc_time, gc_count, example_count,
```

Comment:
Assignment Branch Condition size for dump_summary_to_output is too high. [17.26/15]